src/web/client/dal/fileSystemProvider.ts

```diff
@@ -74,19 +75,27 @@ export class PortalsFS implements vscode.FileSystemProvider {
     // --- manage file metadata
 
     async stat(uri: vscode.Uri): Promise<vscode.FileStat> {
-        const isVersionControlEnabled = vscode.workspace
-            .getConfiguration(SETTINGS_EXPERIMENTAL_STORE_NAME)
-            .get(VERSION_CONTROL_FOR_WEB_EXTENSION_SETTING_NAME);
+        if (fileHasDirtyChanges(uri.fsPath)) {
+            WebExtensionContext.telemetry.sendInfoTelemetry(
+                telemetryEventNames.WEB_EXTENSION_FILE_HAS_DIRTY_CHANGES
+            );
+
+            const isVersionControlEnabled = vscode.workspace
+                .getConfiguration(SETTINGS_EXPERIMENTAL_STORE_NAME)
+                .get(VERSION_CONTROL_FOR_WEB_EXTENSION_SETTING_NAME);
+
+            if (!isVersionControlEnabled) {
+                WebExtensionContext.telemetry.sendInfoTelemetry(
+                    telemetryEventNames.WEB_EXTENSION_DIFF_VIEW_FEATURE_FLAG_DISABLED
```

Comment:
Why we even want to log this? This looks redundant when we make it enabled.
Is it inferring to something which I am missing

Reply:
ideally we would want to know that there are users who explicitly disabled this feature - specially since we are rolling this behind a feature flag to allow the flexibility to disable